From ad1b755fd5594ec348ce58aa49eea1e19edc873f Mon Sep 17 00:00:00 2001
From: Weilong Gao <gwl@rock-chips.com>
Date: Tue, 6 Sep 2016 10:12:12 +0800
Subject: [PATCH] net: rkwifi: add 'disable_proptx = 1' for AP6354

This patch add to fix AP6354's throughput abnormal in p2p mode.

Change-Id: I8c52981dfacee092493442d3223066f9d82c0334
Signed-off-by: Weilong Gao <gwl@rock-chips.com>
---
 .../net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c    | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c
index a827f3012845..07bcc3cc3337 100755
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c
@@ -2380,11 +2380,14 @@ dhd_conf_preinit(dhd_pub_t *dhd)
 	conf->txctl_tmo_fix = FALSE;
 	conf->tx_in_rx = TRUE;
 	conf->rsdb_mode = -2;
-	conf->txglom_mode = SDPCM_TXGLOM_MDESC;
+	conf->txglom_mode = SDPCM_TXGLOM_CPY;
 	if ((conf->chip == BCM43362_CHIP_ID) || (conf->chip == BCM4330_CHIP_ID)) {
 		conf->disable_proptx = 1;
 		conf->use_rxchain = 0;
 	}
+	if (conf->chip == BCM4354_CHIP_ID) {
+		conf->disable_proptx = 1;
+	}
 	if (conf->chip == BCM43430_CHIP_ID) {
 		conf->bus_rxglom = FALSE;
 		conf->use_rxchain = 0;
-- 
2.35.3

